tweak the core-api watch script generation to avoid lot of traces

### Task content

When starting in watch mode I see quite often the log

```
✘ [ERROR] Failed to resolve entry for package "@podman-desktop/core-api". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-scan]

    node_modules/esbuild/lib/main.js:1217:21:
      1217 │         let result = await callback({
           ╵                      ^

```

I think the watch script is not waiting 'enough' the core-api generation.

Tweaking the promise being returned by the core-api package generation may probably help to avoid to see such traces

```
await setupCoreApiPackageWatcher(viteDevServer);
```

I guess we're not waiting the right thing there

--- a/scripts/watch.mjs
+++ b/scripts/watch.mjs
@@ -56,10 +56,24 @@ const stderrFilterPatterns = [
  * @param {{name: string; configFile: string; writeBundle: import('rollup').OutputPlugin['writeBundle'] }} param0
  */
 const getWatcher = ({ name, configFile, writeBundle }) => {
-  return build({
-    ...sharedConfig,
-    configFile,
-    plugins: [{ name, writeBundle }],
+  return new Promise((resolve, reject) => {
+    let resolved = false;
+    build({
+      ...sharedConfig,
+      configFile,
+      plugins: [
+        {
+          name,
+          writeBundle(...args) {
+            writeBundle?.(...args);
+            if (!resolved) {
+              resolved = true;
+              resolve();
+            }
+          },
+        },
+      ],
+    }).catch(reject);
   });
 };
 

